examples/shapes/src/main.rs

```diff
@@ -0,0 +1,194 @@
+                                .as_mut()
+                                .unwrap()
+                                .render_to_surface_async(
+                                    &device_handle.device,
+                                    &device_handle.queue,
+                                    &scene,
+                                    &surface_texture,
+                                    &render_params,
+                                ),
+                        )
+                        .expect("failed to render to surface");
+                        surface_texture.present();
+                        device_handle.device.poll(wgpu::Maintain::Poll);
+                    }
+                    _ => {}
+                }
+            }
+            Event::Suspended => {
+                if let Some(render_state) = render_state.take() {
+                    cached_window = Some(render_state.window);
+                }
+                event_loop.set_control_flow(ControlFlow::Wait);
+            }
+            Event::Resumed => {
+                let Option::None = render_state else { return };
+                // Get the window
+                let window = cached_window
+                    .take()
+                    .unwrap_or_else(|| create_window(event_loop));
+                let size = window.inner_size();
+                let surface_future =
+                    render_cx.create_surface(window.clone(), size.width, size.height);
+                // Create a surface
+                let surface = pollster::block_on(surface_future).expect("Error creating surface");
+                render_state = {
+                    let render_state = RenderState { window, surface };
+                    renderers.resize_with(render_cx.devices.len(), || None);
+                    let id = render_state.surface.dev_id;
+                    renderers[id].get_or_insert_with(|| {
+                        Renderer::new(
+                            &render_cx.devices[id].device,
+                            RendererOptions {
+                                surface_format: Some(render_state.surface.format),
+                                use_cpu: false,
+                                antialiasing_support: vello::AaSupport::all(),
+                                num_init_threads: NonZeroUsize::new(1),
+                            },
+                        )
+                        .expect("Could create renderer")
+                    });
+                    Some(render_state)
+                };
+                event_loop.set_control_flow(ControlFlow::Poll);
+            }
+            _ => {}
+        })
+        .expect("Couldnt run event loop");
+}
+
+fn main() -> Result<()> {
+    // Create the event loop and the rendex context
+    let event_loop = EventLoop::new()?;
+    let render_cx = RenderContext::new().unwrap();
+
+    // Run the event loop
+    run(event_loop, render_cx);
```

Comment:
Since there are no other platforms supported here, I would consider inlining the run function

Reply:
You've inlined the arguments to the run function, but I meant was suggesting inlining the function itself

Reply:
So just remove the `run` function and put all the code in `main`?